src/test/java/org/folio/tlib/api/ApiIT.java

```diff
@@ -94,12 +95,30 @@ public void health() {
 
   @Test
   public void useOverTime() {
-    given().
-      body("{ \"module_to\": \"99.99.99\" }").
-    when().
-      post("/_/tenant").
-    then().
-      statusCode(201);
+    String location =
+        given().
+          body("{ \"module_to\": \"99.99.99\" }").
+        when().
+          post("/_/tenant").
+        then().
+          statusCode(201).
+        extract().
+          header("Location");
+
+    int i = 0;
+    boolean complete;
+    do {
+      if (++i >= 30) {
+        throw new AssertionFailedError("POST /_/tenant timeout");
+      }
+      complete =
+          when().
+            get(location + "?wait=1").
+          then().
+            statusCode(200).
+          extract().
+            path("complete");
+    } while (! complete);
 
```

Comment:
wait=30 waits will only wait as long as it needs to ..

https://github.com/folio-org/mod-eusage-reports/blob/84ee318bb6ac794009e87cb54d8d3111d7513596/src/main/java/org/folio/tlib/api/Tenant2Api.java#L73

so you don't need a loop. ``?wait=30`` should suffice

Reply:
Yes, after fixing the milliseconds vs seconds bug: https://github.com/folio-org/mod-eusage-reports/pull/49/commits/5c643076870fa41cc0b567c4135c35bbee033e34